pix3lify.sh

```diff
@@ -0,0 +1,436 @@
+  if [ "$ABILONG" = "arm64-v8a" ]; then ARCH=arm64; ARCH32=arm; IS64BIT=true; fi;
+  if [ "$ABILONG" = "x86_64" ]; then ARCH=x64; ARCH32=x86; IS64BIT=true; fi;
+}
+
+magisk_version() {
+  if grep MAGISK_VER /data/adb/magisk/util_functions.sh; then
+		echo "$MAGISK_VERSION $MAGISK_VERSIONCODE" >> $ALOG 2>&1
+	else
+		echo "Magisk not installed" >> $ALOG 2>&1
+	fi
+}
+
+# Device Info
+# BRAND MODEL DEVICE API ABI ABI2 ABILONG ARCH
+BRAND=$(getprop ro.product.brand)
+MODEL=$(getprop ro.product.model)
+DEVICE=$(getprop ro.product.device)
+ROM=$(getprop ro.build.display.id)
+api_level_arch_detect
+# Version Number
+VER=$(grep_prop version $MODPROP)
+# Version Code
+REL=$(grep_prop versionCode $MODPROP)
+# Author
+AUTHOR=$(grep_prop author $MODPROP)
+# Mod Name/Title
+MODTITLE=$(grep_prop name $MODPROP)
+#Grab Magisk Version
+MAGISK_VERSION=$(echo $(get_file_value /data/adb/magisk/util_functions.sh "MAGISK_VERSION=") | sed 's|-.*||')
+MAGISK_VERSIONCODE=$(echo $(get_file_value /data/adb/magisk/util_functions.sh "MAGISK_VERSIONCODE=") | sed 's|-.*||')
+
+# Colors
+G='\e[01;32m'		# GREEN TEXT
+R='\e[01;31m'		# RED TEXT
+Y='\e[01;33m'		# YELLOW TEXT
+B='\e[01;34m'		# BLUE TEXT
+V='\e[01;35m'		# VIOLET TEXT
+Bl='\e[01;30m'		# BLACK TEXT
+C='\e[01;36m'		# CYAN TEXT
+W='\e[01;37m'		# WHITE TEXT
+BGBL='\e[1;30;47m'	# Background W Text Bl
+N='\e[0m'			# How to use (example): echo "${G}example${N}"
+loadBar=' '			# Load UI
+# Remove color codes if -nc or in ADB Shell
+[ -n "$1" -a "$1" == "-nc" ] && shift && NC=true
+[ "$NC" -o -n "$LOGNAME" ] && {
+	G=''; R=''; Y=''; B=''; V=''; Bl=''; C=''; W=''; N=''; BGBL=''; loadBar='=';
+}
+
+# Divider (based on $MODTITLE, $VER, and $REL characters)
+character_no=$(echo $MODTITLE $VER $REL | tr " " '_' | wc -c)
+div="${Bl}$(printf '%*s' "${character_no}" '' | tr " " "=")${N}"
+
+# Title Div
+title_div() {
+	no=$(echo $@ | wc -c)
+	extdiv=$((no-character_no))
+	echo "${W}$@${N} ${Bl}$(printf '%*s' "$extdiv" '' | tr " " "=")${N}"
+}
+
+# https://github.com/fearside/ProgressBar
+ProgressBar() {
+# Process data
+	_progress=$(((${1}*100/${2}*100)/100))
+	_done=$(((${_progress}*4)/10))
+	_left=$((40-$_done))
+# Build progressbar string lengths
+	_done=$(printf "%${_done}s")
+	_left=$(printf "%${_left}s")
+
+# 1.2 Build progressbar strings and print the ProgressBar line
+# 1.2.1 Output example:
+# 1.2.1.1 Progress : [########################################] 100%
+printf "\rProgress : ${BGBL}|${N}${_done// /${BGBL}$loadBar${N}}${_left// / }${BGBL}|${N} ${_progress}%%"
+}
+
+# test_connection
+test_connection() {
+	echo -n "Testing internet connection "
+	ping -q -c 1 -W 1 google.com >/dev/null 2>/dev/null && echo "- OK" || { echo "Error"; false; }
+}
+
+# Log files will be uploaded to termbin.com
+upload_logs() {
+	$BBok && {
+		test_connection
+		[ $? -ne 0 ] && exit
+		logup=none;
+		echo "Uploading logs"
+		[ -s $XZLOG ] && logup=$(cat $XZLOG | curl -i -F "Pix3lify_logs.tar.xz"  http://logs.pix3lify.com/submit)
```

Comment:
This has to be `curl -i -F "logs=@Pix3lify_logs.tar.xz"  http://logs.pix3lify.com/submit`. I will work on the `.tar.xz` support.

Reply:
Actually, drop the `-i` you just need the key that my server responses with.

Reply:
Sounds good. I've got all the changes, just trying to test. Magisk is screwing up on me. Got a black screen during install of a different module and now I lost all my installed modules. So trying to get magisk back working properly so I can test

Reply:
Ya I just had something similar happen. Had to factory reset.